### DISCUSSION TITLE
Docker-Container with Ubuntu?

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#17147] [Archive]
I am trying to create a ML server with Directus on top. Since I need packages from python like seaborn and matplotlib it is hard to install them on the Alpine Linux. I am trying to create a Dockerfile which copies the content from the latest Alpine Linux to the Builder.

I am geeting errors like:  ENOENT: no such file or directory, stat '/directus/directus-9.22.4.tgz'

Someone got an idea?

```
FROM directus/directus:latest as builder

FROM sitespeedio/node:ubuntu-20.04-nodejs-12.14.1

COPY --from=builder /directus /directus
COPY --from=builder /directus /directus/directus

#RUN apt-get update && apt-get install -y python3 make g++

ENV TZ=Europe/Minsk
ARG DEBIAN_FRONTEND=noninteractive

RUN apt-get update
RUN apt-get install -y python3 make g++
#RUN apt-get install -y npm
#RUN npm install -g node-pre-gyp

WORKDIR /directus

#RUN npm install

#CMD ["sh", "-c", "node ./cli.js bootstrap && node ./dist/start.js;"]
#CMD ["sh", "-c", "ls nodes"]
CMD ["sh", "-c", "ls -l && npm install && npx directus bootstrap && echo 'Node' && node node_modules/directus/dist/start.js"]
EXPOSE 8055/tcp

```

**Comment by NilsBaumgartner1994** [comment]
I found the main Dockerfile which was not for testing, but i still have no clue.
https://github.com/directus/directus/blob/main/docker/Dockerfile

**Comment by paescuj** [comment]
> I am trying to create a Dockerfile which copies the content from the latest Alpine Linux to the Builder.

Uh, I wouldn't recommend that. Alpine and Ubuntu are two very different systems (for example, Alpine is based on `musl libc` while Ubuntu uses common `glibc`).

> Since I need packages from python like seaborn and matplotlib

Have you already tried to install those packages in Alpine? Apparently, these are available: https://pkgs.alpinelinux.org/package/edge/community/x86/py3-seaborn

If you really need to use Ubuntu, I'd recommend to build things from scratch.

**Reply by NilsBaumgartner1994** [reply]
> > I am trying to create a Dockerfile which copies the content from the latest Alpine Linux to the Builder.
> 
> Uh, I wouldn't recommend that. Alpine and Ubuntu are two very different systems (for example, Alpine is based on `musl libc` while Ubuntu uses common `glibc`).
> 
> > Since I need packages from python like seaborn and matplotlib
> 
> Have you already tried to install those packages in Alpine? Apparently, these are available: https://pkgs.alpinelinux.org/package/edge/community/x86/py3-seaborn
> 
> If you really need to use Ubuntu, I'd recommend to build things from scratch.

How could a Dockerfile look like which makes an exact copy of the directus image, in which i then can install may packes?

**Reply by paescuj** [reply]
Should be pretty straightforward (and similar to what you already had above):
```dockerfile
# Extend from directus image
FROM directus/directus:9.22.4

# Switch to root user
USER root

# Install desired packages
RUN apk --no-cache add \
  py3-seaborn

# Switch back to node user
USER node
```

**Reply by NilsBaumgartner1994** [reply]
Ah thanks. My Problem was that in the latest version i received the error:
```
Error: Cannot find module '/directus/node_modules/directus/dist/storage.js'
```

Which was because an extension is using this. But thanks anyway :-)
